### PR TITLE
encode for py3

### LINKED ISSUE #40434: Python 3 - Head of develop - Salt-ssh not working
Salt-ssh with python3 on head of develop fails 

### Setup
Install salt with python 3.
Fill in relevant info in the roster file.

### Steps to Reproduce Issue
Run salt-ssh command.
`salt-ssh -c ./etc/salt '*' -i test.ping`
```
(virtualenv) [root@li451-148 virtualenv]# salt-ssh -c ./etc/salt '*' -i test.ping
Permission denied for host ssh, do you want to deploy the salt-ssh key? (password required):
[Y/n] y
Password for root@ssh: 
[ERROR   ] An un-handled exception was caught by salt's global exception handler:
TypeError: a bytes-like object is required, not 'str'
Traceback (most recent call last):
  File "/usr/local/bin/salt-ssh", line 6, in <module>
    exec(compile(open(__file__).read(), __file__, 'exec'))
  File "/root/salt/scripts/salt-ssh", line 10, in <module>
    salt_ssh()
  File "/root/salt/salt/scripts.py", line 416, in salt_ssh
    client.run()
  File "/root/salt/salt/cli/ssh.py", line 21, in run
    ssh.run()
  File "/root/salt/salt/client/ssh/__init__.py", line 636, in run
    ret = self.key_deploy(host, ret)
  File "/root/salt/salt/client/ssh/__init__.py", line 358, in key_deploy
    return self._key_deploy_run(host, target, True)
  File "/root/salt/salt/client/ssh/__init__.py", line 381, in _key_deploy_run
    stdout, stderr, retcode = single.shell.copy_id()
  File "/root/salt/salt/client/ssh/shell.py", line 219, in copy_id
    stdout, stderr, retcode = self._run_cmd(self._copy_id_str_new())
  File "/root/salt/salt/client/ssh/shell.py", line 387, in _run_cmd
    term.sendline(self.passwd)
  File "/root/salt/salt/utils/vt.py", line 294, in sendline
    return self.send('{0}{1}'.format(data, linesep))
  File "/root/salt/salt/utils/vt.py", line 288, in send
    return self._send(data)
  File "/root/salt/salt/utils/vt.py", line 569, in _send
    written = os.write(self.child_fd, data)
TypeError: a bytes-like object is required, not 'str'
Traceback (most recent call last):
  File "/usr/local/bin/salt-ssh", line 6, in <module>
    exec(compile(open(__file__).read(), __file__, 'exec'))
  File "/root/salt/scripts/salt-ssh", line 10, in <module>
    salt_ssh()
  File "/root/salt/salt/scripts.py", line 416, in salt_ssh
    client.run()
  File "/root/salt/salt/cli/ssh.py", line 21, in run
    ssh.run()
  File "/root/salt/salt/client/ssh/__init__.py", line 636, in run
    ret = self.key_deploy(host, ret)
  File "/root/salt/salt/client/ssh/__init__.py", line 358, in key_deploy
    return self._key_deploy_run(host, target, True)
  File "/root/salt/salt/client/ssh/__init__.py", line 381, in _key_deploy_run
    stdout, stderr, retcode = single.shell.copy_id()
  File "/root/salt/salt/client/ssh/shell.py", line 219, in copy_id
    stdout, stderr, retcode = self._run_cmd(self._copy_id_str_new())
  File "/root/salt/salt/client/ssh/shell.py", line 387, in _run_cmd
    term.sendline(self.passwd)
  File "/root/salt/salt/utils/vt.py", line 294, in sendline
    return self.send('{0}{1}'.format(data, linesep))
  File "/root/salt/salt/utils/vt.py", line 288, in send
    return self._send(data)
  File "/root/salt/salt/utils/vt.py", line 569, in _send
    written = os.write(self.child_fd, data)
TypeError: a bytes-like object is required, not 'str'
```

### Versions Report
```
salt --versions
Salt Version:
           Salt: 2016.11.3-4139-gde3d585
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.9.5
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 3.5.0 (default, Mar 30 2017, 18:14:57)
   python-gnupg: Not Installed
         PyYAML: 3.12
          PyZMQ: 16.0.2
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.4.3
            ZMQ: 4.1.6
 
System Versions:
           dist: centos 7.3.1611 Core
        machine: x86_64
        release: 4.9.7-x86_64-linode80
         system: Linux
```

### PR DESCRIPTION
**Pull request by gtmanfred**
### What does this PR do?
Encode to bytes so this works for python3

### What issues does this PR fix or reference?
Fixes #40434

### Tests written?

No

### CODE REVIEW

File: salt/utils/vt.py [hunk 1]
```diff
@@ -566,7 +566,7 @@ def _send(self, data):
             try:
                 if self.stdin_logger:
                     self.stdin_logger.log(self.stdin_logger_level, data)
-                written = os.write(self.child_fd, data)
+                written = os.write(self.child_fd, data.encode('utf-8'))
```
Comment (s0undt3ch) [inline salt/utils/vt.py]: Use `__salt_system_encoding__` instead.

Comment (s0undt3ch) [inline salt/utils/vt.py]: And make sure you do that for py3 only.

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

File: salt/utils/vt.py (hunk 1 above)
Comment (s0undt3ch) [inline salt/utils/vt.py]: Or we'll break py2 compat.

Comment (gtmanfred) [inline salt/utils/vt.py]: I tested on py2, and it was working on py2 :/

**Review by s0undt3ch — APPROVED** [review]

**Comment by s0undt3ch** [comment]
Missing the six import.

**Review by s0undt3ch — APPROVED** [review]

**Pull request merged**